File: examples/hello.rb
# Ruby test script
puts "Hello from Ruby!"
puts "Running on Ditto embedded interpreter"

# Variables
name = "Ditto"
version = "0.1.0"

# String interpolation
puts "Project: #{name} v#{version}"

# Arithmetic
a = 10
b = 20
puts "Sum: #{a + b}"

# Array
colors = ["red", "green", "blue"]
puts "Colors: #{colors.length} items"

# Each loop
colors.each do |color|
  puts "  - #{color}"
end

# Times loop
3.times do
  puts "Counting..."
end

# If statement
x = 10
puts "x is big" if x > 5

# Method definition
def greet(person)
  puts "Hello, #{person}!"
end

greet("World")

File: examples/hello.go
package main

import "fmt"

func main() {
	// Basic output
	fmt.Println("Hello from Go!")
	fmt.Println("Running on Ditto embedded interpreter")

	// Variables
	name := "Ditto"
	version := "0.1.0"
	fmt.Println("Project:", name, "v"+version)

	// Arithmetic
	a := 10
	b := 20
	fmt.Println("Sum:", a+b)

	// Array
	colors := []string{"red", "green", "blue"}
	fmt.Println("Colors:", len(colors), "items")

	// Loop
	for i := 0; i < 3; i++ {
		fmt.Println("Counting:", i)
	}

	// If statement
	x := 10
	if x > 5 {
		fmt.Println("x is greater than 5")
	}
}
